modules/application.py

```diff
@@ -360,13 +363,16 @@ def removeAddon(self):
             for item in contents:
                 itemDir = "{}/{}".format(parent, item)
                 if os.path.isdir(itemDir) and not item.lower().startswith("blizzard_"):
-                    toc = "{}/{}.toc".format(itemDir, item)
-                    if os.path.exists(toc):
-                        tmp = self.extractAddonMetadataFromTOC(toc)
-                        if tmp[0] == addonName:
-                            rmtree(itemDir)
-                            deleted_addons.append(item)
-                            deleted = True
+                    for files in os.listdir(itemDir):
+                        toc = "{}/{}.toc".format(itemDir, item)
+                        if files.find(".toc") != -1:
+                            toc = "{}/{}".format(itemDir, files)
```

Comment:
here too. I think search for a file in the files ending with .toc would be enough and use this.